### PR TITLE
[Core] Fix ScrollToPosition.MakeVisible bug

### PR DESCRIPTION
**Pull request by myroot**
### Description of Change ###
 Fix ScrollToPosition.MakeVisible bug
 - Fix `isItemVisible` condition on ScrollView

### Bugs Fixed ###
As is
Image Image

Fixed
Image Image

### API Changes ###
None

### Behavioral Changes ###
None

### PR Checklist ###

- [x] Has tests (if omitted, state reason in description)
- [x] Rebased on top of the target branch at time of PR
- [x] Changes adhere to coding standard
- [x] Consolidate commits as makes sense

### CODE REVIEW

File: Xamarin.Forms.Core/ScrollView.cs [hunk 1]
```diff
@@ -102,7 +102,7 @@ public Point GetScrollPositionForElement(VisualElement item, ScrollToPosition po
 
 			if (position == ScrollToPosition.MakeVisible)
 			{
-				bool isItemVisible = ScrollX < y && ScrollY + Height > y;
+				bool isItemVisible = ScrollX < x && ScrollX + Width > x + item.Width && ScrollY < y && ScrollY + Height > y + item.Height;
```
Comment (paymicro) [inline Xamarin.Forms.Core/ScrollView.cs]: This very difficult to read.
I think that it should be divided into several parts.
```C#
bool isVisibleHorizontal = x < ScrollX && ScrollX > x + item.Width - Width;
bool isVisibleVertical = y < ScrollY && ScrollY > y + item.Height - Height;
bool isItemVisible = isVisibleHorizontal && isVisibleVertical;
```

Comment (rmarinho) [reply Xamarin.Forms.Core/ScrollView.cs]: agree with @paymicro let's make that change

**Comment by myroot** [comment]
@rmarinho @paymicro 
Thanks for the review, I update for code readability

**Comment by StephaneDelcroix** [comment]
could you rebase your branch on top of 3.1.0 ? thanks

**Comment by myroot** [comment]
@StephaneDelcroix Do you want to rebase to 3.1.0 branch? instead master? Ok

**Comment by myroot** [comment]
I have done rebase to 3.1.0 branch

**Comment by rmarinho** [comment]
It's failing some UITests on iOS11 , i redeploy the test run to check

**Comment by myroot** [comment]
@rmarinho Can I ask reason to fail the UITests? I'm wonder what has to do to be merged

**Comment by rmarinho** [comment]
@myroot sure:

``` 
IOS11 

Xamarin.Forms.Core.UITests.ScrollViewGalleryTests.ScrollToElement1Start_apple_iphone_x-11_3_1

System.Exception : Error while performing WaitForElement(Marked("the scrollto button"), "Timed out waiting for element...", null, null, null) 

Xamarin.Forms.Core.UITests.ScrollViewGalleryTests.ScrollToElement2Center_apple_iphone_x-11_3_1
Xamarin.Forms.Core.UITests.ScrollViewGalleryTests.ScrollToElement3End_apple_iphone_x
```

**Comment by myroot** [comment]
@rmarinho Thanks for information. I now know reason to fail because i modify ScrollViewGallery.
I will rollback ScrollViewGallery soon.

**Pull request merged**